### PR TITLE
Allow ignoring out of order exemplars

### PR DESCRIPTION
**Pull request by krajorama**
#### What this PR does

Out of order examplars are not supported in Prometheus and Mimir. Tipical Prometheus remote write and OTLP clients cannot gracefully handle the partial error that Mimir returns when it encounters an out of order exemplar. Add a per tenant configuration option to silence the error towards the client.

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [X] Tests updated.
- [N/A] Documentation added.
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [N/A] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Comment by bboreham** [comment]
Linking upstream https://github.com/prometheus/prometheus/issues/13577

File: pkg/ingester/ingester.go [hunk 1]
```diff
@@ -1597,10 +1598,12 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 						}
 					}
 
-					// Error adding exemplar
-					updateFirstPartial(nil, func() softError {
-						return newTSDBIngestExemplarErr(err, model.Time(ex.TimestampMs), ts.Labels, ex.Labels)
-					})
+					// Error adding exemplar. Do not report to client if the error was out of order and we ignore such error.
+					if !(isOOOExemplar && i.limits.IgnoreOOOExemplars(userID)) {
```
Comment (pstibrany) [inline pkg/ingester/ingester.go:1602]:
```suggestion
					if !isOOOExemplar || !i.limits.IgnoreOOOExemplars(userID) {
```

**Review by pstibrany — APPROVED** [review]
thank you!

**Review by dimitarvdimitrov — COMMENTED** [review]
nice!

**Pull request merged**